xarray/coding/cftime_offsets.py

```diff
@@ -0,0 +1,765 @@
+
+    +----------------------+-------------------------------------------------------------------------+
+    | Alias                | Date type                                                               |
+    +======================+=========================================================================+
+    | standard             | ``np.datetime64`` (falls back to ``cftime.DatetimeProlepticGregorian``) |
+    +----------------------+-------------------------------------------------------------------------+
+    | gregorian            | ``cftime.DatetimeGregorian``                                            |
+    +----------------------+-------------------------------------------------------------------------+
+    | proleptic_gregorian  | ``cftime.DatetimeProlepticGregorian``                                   |
+    +----------------------+-------------------------------------------------------------------------+
+    | noleap, 365_day      | ``cftime.DatetimeNoLeap``                                               |
+    +----------------------+-------------------------------------------------------------------------+
+    | all_leap, 366_day    | ``cftime.DatetimeAllLeap``                                              |
+    +----------------------+-------------------------------------------------------------------------+
+    | 360_day              | ``cftime.Datetime360Day``                                               |
+    +----------------------+-------------------------------------------------------------------------+
+    | julian               | ``cftime.DatetimeJulian``                                               |
+    +----------------------+-------------------------------------------------------------------------+
+
+    The ``CFTimeIndex``-enabled portion of this function supports most of the
+    features of ``pandas.date_range`` (e.g. specifying how the index is
+    ``closed`` on either side, or whether or not to ``normalize`` the start and
+    end bounds).  Notable exceptions, however, are:
+
+    - You cannot specify a ``tz`` (time zone) argument for a date range that
+      results in a ``CFTimeIndex``.
+    - Start or end dates specified as partial-datetime strings must use the
+      `ISO-8601 format <https://en.wikipedia.org/wiki/ISO_8601>`_.
+    - The ``CFTimeIndex``-enabled version of the function supports many, but
+      not all frequencies supported by the ``DatetimeIndex`` version.  For
+      example it does not currently support any of the business-related,
+      semi-monthly, or sub-second frequencies.
+    - Compound sub-monthly frequencies are not supported, e.g. '1H1min', as
+      these can easily be written in terms of the finest common resolution,
+      e.g. '61min'.
+
+    Examples
+    --------
+
+    For dates from standard calendars within the ``pandas.Timestamp``-valid
+    range, this function operates as a thin wrapper around
+    ``pandas.date_range``.
```

Comment:
I don't think this use-case is important to support, and interfaces that can return different types can be difficult to understand. I would instead always return a `CFTimeIndex`.